plugin/mothership.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash
+
+
+source $KITBAG_HOME/plugin/k3s.sh
+
+# Public configuration (adjustable over CLI)
+MOTHERSHIP_OPTIONS="
+MOTHERSHIP_CORRELATION_ID|corelation_id|
+MOTHERSHIP_RUNTIME_ID|runtime_id|
+MOTHERSHIP_SCHEDULING_ID|scheduling_id|
+MOTHERSHIP_SHOOT|shoot|
+MOTHERSHIP_EXCLUDE|exclude|
+MOTHERSHIP_KUBECONFIG|kubeconfig|
+MOTHERSHIP_K8S_NAME|kcn|msrec
+"
+
+#
+# Hook executed when the PLUGIN was loaded.
+#
+function _mothership_main {
+  cmdExists docker
+  cmdExists k3d
+  cmdExists istioctl
+  cmdExists msrec
+  cmdExists kcp
+  cmdExists jq
+  cmdExists tr
+
+  varNotEmpty MOTHERSHIP_KUBECONFIG true
+  fileExists ${MOTHERSHIP_KUBECONFIG} true
+  
+  MOTHERSHIP_KCP_OPTION=$(_mothership_validate_exclusive_args \
+	  ${!MOTHERSHIP_CORRELATION_ID@} \
+	  ${!MOTHERSHIP_RUNTIME_ID@} \
+	  ${!MOTHERSHIP_SCHEDULING_ID@} \
+	  ${!MOTHERSHIP_SHOOT@})
+
+  varNotEmpty ISTIOCTL_PATH true
+  varNotEmpty KCPCONFIG true
+
+  MOTHERSHIP_LOCAL_JQ_FILTER=$(_mothership_parse_excluded)
+  
+  # create a temp mothership reconciler configuration file
+  MOTHERSHIP_TEMP_MSRECONFIG=$(mktemp /tmp/msreconfig-XXXXX)
```

Comment:
Wouldn`t it make more sense to put such validation in the `mothership_local` command. Thus, it gets only executed when `kitbag mothership local` gets called?

For example when a user wants to know what the reconciler command can do, without having set the required arguments, then he/seh would call `kitbag mothership help` - which leads to the following:
```
./kitbag mothership

ERROR:
    * Command 'msrec' not found in path
    * The variable 'MOTHERSHIP_KUBECONFIG' was not defined or is empty


Usage:
kitbag  <command> [-optionKey optionValue]*

Supported commands for '' are:
  * self            (sub-plugin)
  * golang          (sub-plugin)
  * mothership      (sub-plugin)
  * k3s             (sub-plugin)

Available options for 'mothership' are [Name|Alias|DefaultValue|Description]:
  * MOTHERSHIP_CORRELATION_ID|corelation_id|
  * MOTHERSHIP_RUNTIME_ID|runtime_id|
  * MOTHERSHIP_SCHEDULING_ID|scheduling_id|
  * MOTHERSHIP_SHOOT|shoot|
  * MOTHERSHIP_EXCLUDE|exclude|
  * MOTHERSHIP_KUBECONFIG|kubeconfig|
  * MOTHERSHIP_K8S_NAME|kcn|msrec
  * help
```

The user should not get an error message, when calling help. Thus I would move it to the public functions

Reply:
- https://github.com/kyma-incubator/kitbag/issues/11
- https://github.com/kyma-incubator/kitbag/issues/12